### DISCUSSION TITLE
[x-intersect:leave] Ambiguous Documentation

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#4027] [4. General]
https://alpinejs.dev/plugins/intersect#x-intersect-leave states that it is triggered when the element leaves the viewport. However, leaving the viewport can mean two things:

1. **Part** of an element has left the viewport
2. The **whole element** has left the viewport _(currently implemented as far as I can tell)_

I would suggest to define it more specific and optionally also make (1) available as well.

**Comment by SimoTod** [comment]
If you scroll to the bottom of that page, there are modifiers to change the behaviour.
The documentation is open source so feel free to submit improvements (same repo > https://github.com/alpinejs/alpine/tree/main/packages/docs/src/en).

**Comment by simonferndriger** [comment]
Thanks - do you mean https://alpinejs.dev/plugins/intersect#threshold specifically? In the doc there is only the entering into the viewport mentioned, which got me thinking that this can't be used for leaving the viewport.

So, did I get this correctly, that the default threshold for _entering_ the viewport is `0`, while the default threshold for _leaving_ the viewport is `100`?

**Reply by ekwoka** [reply]
could be clarified that it is working with `IntersectionObserver` and link to the MDN page for that.

> So, did I get this correctly, that the default threshold for entering the viewport is 0, while the default threshold for leaving the viewport is 100

No.

The threshold is basically "how much of the element must be in the viewport for the thing to be counted as intersecting the viewport".

For both entering and leaving.

By default, the threshold is 0 and the margins are 0. So if any of the element is in the viewport, it's counted as "in the viewport".

https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver

**Reply by SimoTod** [reply]
> Thanks - do you mean https://alpinejs.dev/plugins/intersect#threshold specifically? In the doc there is only the entering into the viewport mentioned, which got me thinking that this can't be used for leaving the viewport.
> 
> So, did I get this correctly, that the default threshold for _entering_ the viewport is `0`, while the default threshold for _leaving_ the viewport is `100`?

The documentation was written by the developer who sent the PR. He was focusing on the enter bit but it works for leave as well. Not a great naming choice but x-intersect:leave.full will trigger as soon as the first few pixels are out of the viewport. You can also use threshold if you want a different value.

**Reply by simonferndriger** [reply]
OK, I think I get it now. So `x-intersect:leave.full` does not mean what it suggests (that the element has **full**y left the viewport, but the element is NOT fully in the viewport anymore, thereby leaving with part of the element. Or with other words, the modifiers are always in respect to entering/visible and never directly to leaving (leave = !visible).

A bit confusing but understandable 😊

**Reply by ekwoka** [reply]
Yeah, it's not great. It makes sense based on the underlying Observer, and you kind of just need to think about the modifiers in isolation.

**Comment by simonferndriger** [comment]
Extended the documentation: https://github.com/alpinejs/alpine/pull/4032
